mlflow/tracing/types/model.py

```diff
@@ -0,0 +1,126 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+
+
+@dataclass
+class Trace:
+    trace_info: "TraceInfo"
+    trace_data: "TraceData"
+
+
+@dataclass
+class TraceInfo:
+    trace_id: str
+    # Trace name is inherited from the root span name at the time of trace creation.
+    # This may be edited by the user later.
+    name: str
+    # Start and end time of the trace, inherited from the root span.
+    start_time: int
+    end_time: int
+    # Status of the trace, inherited from the root span.
+    status: "Status"
+    # Input and output data of the root span, but serialized and truncated to fixed
+    # length for the efficient storage and retrieval.
+    inputs: str
+    outputs: str
+    # Metadata should only be set by the system and immutable.
+    metadata: Dict[str, Any]
+    # Tags can be mutated by the user to attach additional information to the trace.
+    tags: Dict[str, Union[str, float]]
+    # Save the entity (model_id, app_version_id, etc) from which the trace is generated
+    source: Optional[str] = None
+
+
+@dataclass
+class TraceData:
+    spans: List["Span"]
+
+
+@dataclass
+class Span:
+    name: str
+    context: "SpanContext"
```

Comment:
+1 to the earlier mention of definitions here... let's use future annotations and direct class names to avoid NameError exceptions instead of string wrapping for deferred resolution.